File: python/examples/export_and_read.py
import os

from memory_cloud import MemoryCloudClient, read_export_package


def main():
    # 1) Create client.
    client = MemoryCloudClient(
        base_url=os.getenv("AWARENESS_API_BASE_URL", os.getenv("AWARENESS_BASE_URL", "http://localhost:8000/api/v1")),
        api_key="YOUR_API_KEY",
    )

    # 1) Export safetensors package with optional on-export KV generation.
    export_path = client.save_export_memory_package(
        memory_id="your-memory-id",
        payload={
            "package_type": "vector_with_kv_cache",
            "vector_binary_format": "safetensors",
            "include_kv_summary": True,
            "include_kv_binary": False,
            "generate_kv_on_export": True,
            "kv_latent_steps": 3,
            "regenerate_vectors_if_missing": True,
        },
        output_path="memory_export.zip",
    )
    print("saved:", export_path)

    # 2) Read export zip from SDK helper.
    parsed = read_export_package(export_path)
    manifest = parsed.get("manifest", {})
    print("package_type:", manifest.get("export", {}).get("package_type"))
    print("vector_format:", manifest.get("export", {}).get("vector_binary_format"))
    print("chunks:", len(parsed.get("chunks") or []))
    print("vectors_jsonl:", len(parsed.get("vectors_jsonl") or []))
    print("has_safetensors:", bool(parsed.get("safetensors")))
    print("kv_summary_keys:", list((parsed.get("kv_summary") or {}).keys()))


if __name__ == "__main__":
    main()

File: typescript/examples/export-and-read.ts
import { MemoryCloudClient, readExportPackage } from "../src";
import { writeFile } from "node:fs/promises";

async function main() {
  // 1) Create client.
  const apiBaseUrl = process.env.AWARENESS_API_BASE_URL || "https://awareness.market/api/v1";
  const client = new MemoryCloudClient({
    baseUrl: apiBaseUrl,
    apiKey: "YOUR_API_KEY",
  });

  // 2) Export package.
  const exported = await client.exportMemoryPackage({
    memoryId: "your-memory-id",
    payload: {
      package_type: "vector_with_kv_cache",
      vector_binary_format: "safetensors",
      include_kv_summary: true,
      include_kv_binary: false,
      generate_kv_on_export: true,
      kv_latent_steps: 3,
      regenerate_vectors_if_missing: true,
    },
  });

  await writeFile(exported.filename, exported.bytes);
  console.log("saved:", exported.filename, exported.bytes.byteLength);

  // 3) Parse package directly from bytes.
  const parsed = await readExportPackage(exported.bytes);
  const manifestExport = (parsed.manifest.export ?? {}) as Record<string, unknown>;
  console.log("package_type:", manifestExport.package_type);
  console.log("vector_format:", manifestExport.vector_binary_format);
  console.log("chunks:", parsed.chunks.length);
  console.log("vectorsJsonl:", parsed.vectorsJsonl.length);
  console.log("hasSafetensors:", Boolean(parsed.safetensors));
  console.log("kvSummaryKeys:", Object.keys(parsed.kvSummary ?? {}));
}

main().catch((err) => {
  console.error(err);
  process.exitCode = 1;
});
